src/main/java/com/gluonhq/substrate/util/FileOps.java

```diff
@@ -307,4 +323,48 @@ public static String calculateCheckSum(File file) {
             return "";
         }
     }
+
+    /**
+     * Extracts the files that match a given extension found in a jar to a target patch,
+     * providing that the file passes a given filter, and it doesn't exist yet in the target path
+     *
+     * @param extension the extension of the files in the jar that will be extracted
+     * @param sourceJar the path to the jar that will be inspected
+     * @param target the path of the folder where the files will be extracted
+     * @param filter a predicate that the files in the jar should match.
+     * @throws IOException
+     */
+    public static void extractFilesFromJar(String extension, Path sourceJar, Path target, Predicate<Path> filter) throws IOException {
+        if  (!Files.exists(sourceJar)) {
+            return;
```

Comment:
when will this happen? Doesn't this indicate an exception? Someone asks to extracts native libs from a non-exiting jar?

Reply:
This is just a regular check to bail out if the sourcePath is not a valid path. While that is not expected, this is a method in FileOps that could be used for many different purposes.
I wouldn't throw an exception, but a log can be added.